### PR TITLE
add `contexts` to `script.addPreloadScript`

### LINKED ISSUE #373: Add `browsingContext` parameter to `script.addPreloadScript`
Currently, there is no way for the ["preload script"](https://w3c.github.io/webdriver-bidi/#preload-script) to be run only in specific BrowsingContext, and there is not way for the script to determine in which browsing context it is running, which makes some scenarios quite hard to implement. E.g. scenario "setup mutation observer on a page I'm about to navigate".

I suggest adding either `context: browsingContext.BrowsingContext` or `?contexts: [*browsingContext.BrowsingContext]` parameter to the [`script.addPreloadScript`](https://w3c.github.io/webdriver-bidi/#command-script-addPreloadScript) command.

### PR DESCRIPTION
**Pull request by Lightning00Blade**
Closes #373

***
<a href="https://pr-preview.s3.amazonaws.com/Lightning00Blade/webdriver-bidi/pull/517.html" title="Last updated on Aug 28, 2023, 3:19 PM UTC (8215e30)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webdriver-bidi/517/0329f44...Lightning00Blade:8215e30.html" title="Last updated on Aug 28, 2023, 3:19 PM UTC (8215e30)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -4416,6 +4416,12 @@ To <dfn export>run WebDriver BiDi preload scripts</dfn> given |environment setti
   1. For each |preload script| in |session|'s [=preload script map=]'s
      [=values=]:
 
+    1. If |preload script|'s <code>contexts</code> is not null.
```
Comment (OrKoN) [inline index.bs:4419]:
```suggestion
    1. If |preload script|'s <code>contexts</code> is not null:
```

File: index.bs [hunk 2]
```diff
@@ -4416,6 +4416,12 @@ To <dfn export>run WebDriver BiDi preload scripts</dfn> given |environment setti
...
+
+       1. Let |top level context| be |browsing context|’s [=top-level browsing context=].
+
+       2. If |top level context| is not in <code>contexts</code>, return.
```
Comment (OrKoN) [inline index.bs:4423]: maybe needs `[=set/contains=]` or `[=list/contains=]`?

**Review by OrKoN — APPROVED** [review]
LGTM (wait for other reviewers)

File: index.bs [hunk 3]
```diff
@@ -4416,6 +4416,12 @@ To <dfn export>run WebDriver BiDi preload scripts</dfn> given |environment setti
   1. For each |preload script| in |session|'s [=preload script map=]'s
      [=values=]:
 
+    1. If |preload script|'s <code>contexts</code> is not null:
+
+       1. Let |top level context| be |browsing context|’s [=top-level browsing context=].
+
+       1. If <code>contexts</code> does not [=list/contains|contain=] |top level context|, return.
```
[the remark this answers is not in the document]
Comment (Lightning00Blade) [reply index.bs:4423]: This sound correct, will make this change.

**Review by juliandescottes — APPROVED** [review]
That looks good to me now! Given that @jgraham is away, it would be great if @whimboo or @lutien could do a quick double check.

**Review by thiagowfx — APPROVED** [review]

**Review by lutien — APPROVED** [review]
Looks good to me as well

**Comment by whimboo** [comment]
@Lightning00Blade as usual we need tests for this particular addition, and we should not necessarily have merged it before. Are you planning to write those? Thanks.

**Comment by whimboo** [comment]
@Lightning00Blade do you already know when you could provide these new tests? We would like to implement this feature as well and it would be great to have already existent tests. Thanks.

**Comment by thiagowfx** [comment]
It would be nice to open a separate bug to keep track of missing WPT test implementations.

**Pull request merged**